File: IntroJava/src/SubtractionQuizLoop.java
/*
Listing 5.4 SubtractionQuizLoop.java
Generates five questions and, after a student answers all five,
reports the number of correct answers. The program also displays the time spent on the
test and lists all the questions.

final int NUMBER_OF_FIVE_QUESTIONS = 5
count: 0-4
int numberOfCorrectAnswers
double startTime = System.currentTimeMillis();
 */

import java.util.Scanner;

public class SubtractionQuizLoop
{
    public static void main(String[] args)
    {
        final int NUMBER_OF_FIVE_QUESTIONS = 5;
        int count = 0;
        int numberOfCorrectAnswers = 0;
        double startTime = System.currentTimeMillis();

        Scanner input = new Scanner(System.in);

        while (count < NUMBER_OF_FIVE_QUESTIONS)
        {
            int number1 = (int) (Math.random() * 10);
            int number2 = (int) (Math.random() * 10);
            if (number1 < number2)
            {
                int temp = number1;
                number1 = number2;
                number2 = temp;
            }

            System.out.printf("%d - %d = ", number1, number2);
            int answer = input.nextInt();
            if (number1 - number2 == answer)
            {
                System.out.println("You are right!\n");
                numberOfCorrectAnswers ++;
            }
            else
            {
                System.out.print("You are wrong!\n");
                System.out.printf("%d - %d = %d\n\n", number1, number2, number1 - number2);
            }

            count ++;
        }

        double endTime = System.currentTimeMillis();
        double timeSpent = (endTime - startTime) / 1000;
        System.out.printf("Test time is %d mintue(s) and %d second(s).\n",
                (int) (timeSpent / 60), (int) (timeSpent % 60));

        System.out.printf("Correct count is %d", numberOfCorrectAnswers);
    }
}

File: IntroPython/SubtractionQuizLoop.py
# LISTING 5.4 SubtractionQuizLoop.py

import random
import time

correctCount = 0
count = 0  # Count the number of question
NUMBER_OF_QUESTION = 5

startTime = time.time()

while count < NUMBER_OF_QUESTION:
    number1 = random.randint(0, 9)
    number2 = random.randint(0, 9)
    if number1 < number2:
        number1, number2 = number2, number1
    result = eval(input("What's " + str(number1) + " - " + str(number2) + " = ? "))
    if result == number1 - number2:
        print("You are right!")
        correctCount += 1
    else:
        print("You are wrong!", number1, "-", number2, "=", number1 - number2)

    count += 1

endTime = time.time()
print("Correct count is", correctCount, "out of", NUMBER_OF_QUESTION, \
      "\nTest time is", endTime - startTime, "seconds.")
